### PR TITLE
Cleanup primitive structures

### PR DESCRIPTION
**Pull request by blaxill**
- This seems to be a performance degradation somewhat, I am investigating options

### CODE REVIEW

File: silveroak-opentitan/aes/Arrow/_CoqProject [hunk 1]
```diff
@@ -18,7 +18,6 @@ unrolled_naive_cipher.v
 unrolled_opentitan_cipher.v
 cipher_round.v
 key_expand.v
-cipher_control.v
```
Comment (jadephilipoom) [inline silveroak-opentitan/aes/Arrow/_CoqProject:21]: Is this supposed to be removed?

**Review by jadephilipoom — APPROVED** [review]
LGTM!

File: silveroak-opentitan/aes/Arrow/_CoqProject (hunk 1 above)
Comment (blaxill) [reply silveroak-opentitan/aes/Arrow/_CoqProject:21]: Yeah, type unification is particularly flakey in the combination of this definition and the custom notation it uses. I will introduce a better fix for this in upcoming notational changes

**Comment by jadephilipoom** [comment]
This actually broke `silveroak-opentitan/aes/`; the CI didn't catch it because of the issue mentioned in https://github.com/project-oak/oak-hardware/pull/328#issuecomment-720485345

CI should be able to catch the issue now.

**Pull request merged**